merlin-driver/src/main/java/gov/nasa/jpl/aerie/merlin/driver/SimulationDriver.java

```diff
@@ -110,8 +110,9 @@ public static <Model> SimulationResults simulate(
         // Swallowing the spanException as the internal `spanId` is not user meaningful info.
         final var topics = missionModel.getTopics();
         final var directiveId = engine.getDirectiveIdFromSpan(activityTopic, topics, ex.spanId);
+        final var activityTrace = engine.findActivityTraceFromSpan(engine, activityTopic, topics, ex.spanId);
```

Comment:
Minor, repeated for `CheckpointSimDriver` and `CachedSimEngine`: It looks like the trace should only calculated if `directiveId.isPresent()`, so it ought to go inside the `if` block.